### PR TITLE
Rename progsize to progbufsize.

### PR DESCRIPTION
**Pull request by timsifive**
Also take a stab at explaining the benefit of a few different sizes.

Issue #117.

### CODE REVIEW

File: xml/dm_registers.xml [hunk 1]
```diff
@@ -276,9 +276,19 @@
...
     <register name="Abstract Control and Status" short="abstractcs" address="0x16">
         <field name="0" bits="31:29" access="R" reset="0" />
-        <field name="progsize" bits="28:24" access="R" reset="Preset">
+        <field name="progbufsize" bits="28:24" access="R" reset="Preset">
             Size of the Program Buffer, in 32-bit words. Valid sizes are 0 - 16.
 
+            0: Tiny systems that support System Bus Access might not have a
+            Program Buffer at all.
+
+            1: Tiny systems can support just a single-word program buffer. The
```
Comment (mwachs5) [inline xml/dm_registers.xml]: Putting these here makes it sound like the only options. I think this explanatory text belongs where program buffer is discussed (despite the fact that the TODO Is here).

File: debug_module.tex [hunk 2]
```diff
@@ -245,20 +245,24 @@ \subsection{Abstract Command Listing}
...
+the debugger need not write it. The contents of \Rprogbufzero
+can be a 32-bit
 instruction, or a compressed instruction in the lower 16 bits accompanied by a
-compressed {\tt nop} in the upper 16 bits.  If \Fprogsize is greater than 1,
-the debugger can write whatever program it likes, but the program must end with
-{\tt ebreak} or {\tt ebreak.c}.
+{\tt c.nop} in the upper 16 bits. 
+If \Fprogbufsize is greater than 1,
+the debugger can write any program, but it must
+write an {\tt ebreak} or {\tt c.ebreak} at the end of the program. Failure
```
Comment (timsifive) [inline debug_module.tex]: Remove "an"

File: debug_module.tex [hunk 3]
```diff
@@ -245,20 +245,24 @@ \subsection{Abstract Command Listing}
...
 execute it exactly once with the Access Register Abstract Command,
-setting the \Fpostexec bit in \Rcommand.
-If \Fprogsize is 1, the Program Buffer may only hold a single instruction.
-This can be a 32-bit
+setting the \Fpostexec bit in \Rcommand. The program in the 
+program buffer must end with {\tt ebreak} or {\tt c.ebreak}.
+If \Fprogbufsize is 1, the final {\tt ebreak} is implied and 
+the debugger need not write it. The contents of \Rprogbufzero
+can be a 32-bit
 instruction, or a compressed instruction in the lower 16 bits accompanied by a
```
Comment (timsifive) [inline debug_module.tex]: Remove "a" at the end

File: debug_module.tex [hunk 4]
```diff
@@ -245,20 +245,24 @@ \subsection{Abstract Command Listing}
...
-compressed {\tt nop} in the upper 16 bits.  If \Fprogsize is greater than 1,
-the debugger can write whatever program it likes, but the program must end with
-{\tt ebreak} or {\tt ebreak.c}.
+{\tt c.nop} in the upper 16 bits. 
+If \Fprogbufsize is greater than 1,
```
Comment (timsifive) [inline debug_module.tex]: Isn't this just saying the same as the above "The program in the program buffer must end with {\tt ebreak} or {\tt c.ebreak}."

**Pull request merged**